### DISCUSSION TITLE
How to cache selected data from select in react query?

### DISCUSSION THREAD

**Discussion by deepgolani4** [TanStack/query#4291] [General] [answered]
is there any way to cache only selected data (transformed by `select` props in useQuery) in the query cache?
if no then what are the alternative

**Comment by TkDodo** [comment, marked as answer]
no there is not because select is specific to a per-query basis. Each invocation of `useQuery` could have it's own select, which selects different data.

There are tons of ways to do data transformation, see: https://tkdodo.eu/blog/react-query-data-transformations
